File: objc/Demo Objc/TestValidationViewController.m
//
//  TestValidationViewController.m
//  Demo Objc
//
//  Created by Eduardo Mello de Macedo | Stone on 10/03/17.
//  Copyright © 2017 Eduardo Mello de Macedo | Stone. All rights reserved.
//

#import "TestValidationViewController.h"
#import "NSString+Utils.h"

@interface TestValidationViewController ()

@property (strong, nonatomic) IBOutlet UIButton *activationButton;
@property (strong, nonatomic) IBOutlet UIButton *pinpadConnectionButton;
@property (strong, nonatomic) IBOutlet UIButton *tableDownloadButton;
@property (strong, nonatomic) IBOutlet UIButton *internetConnectionButton;

@end

@implementation TestValidationViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.title = [kTitleValidation localize];
    [self.activationButton setTitle:[kButtonActivation localize] forState:UIControlStateNormal];
    [self.pinpadConnectionButton setTitle:[kButtonPinpadConnection localize] forState:UIControlStateNormal];
    [self.tableDownloadButton setTitle:[kButtonTableDownload localize] forState:UIControlStateNormal];
    [self.internetConnectionButton setTitle:[kButtonInternetConnection localize] forState:UIControlStateNormal];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (IBAction)performActivation:(id)sender {
    if ([STNValidationProvider validateActivation] == YES) {
        NSLog(@"%@", [kLogIsActivated localize]);
        self.feedback.text = [kLogIsActivated localize];
    } else {
        NSLog(@"%@", [kLogNotActivated localize]);
        self.feedback.text = [kLogNotActivated localize];
    }
}

- (IBAction)performPinpadConnection:(id)sender {
    if ([STNValidationProvider validatePinpadConnection] == YES) {
        NSLog(@"%@", [kLogPinpadConnected localize]);
        self.feedback.text = [kLogPinpadConnected localize];
    } else {
        NSLog(@"%@", [kLogPinpadNotConnected localize]);
        self.feedback.text = [kLogPinpadNotConnected localize];
    }
}

- (IBAction)PerfomTablesDownloaded:(id)sender {
    if ([STNValidationProvider validateTablesDownloaded] == YES) {
        NSLog(@"%@", [kLogTablesDownloaded localize]);
        self.feedback.text = [kLogTablesDownloaded localize];
    } else {
        NSLog(@"%@", [kLogTablesNotDownloaded localize]);
        self.feedback.text = [kLogTablesNotDownloaded localize];
    }
}

- (IBAction)performConnectionNetwork:(id)sender {
    if ([STNValidationProvider validateConnectionToNetWork] == YES) {
        NSLog(@"%@", [kLogInternetConnection localize]);
        self.feedback.text = [kLogInternetConnection localize];
    } else {
        NSLog(@"%@", [kLogNoInternetConnection localize]);
        self.feedback.text = [kLogNoInternetConnection localize];
    }
}

@end

File: swift/Demo Swift/Demo Swift/TestValidationViewController.swift
//
//  TestValidationViewController.swift
//  Demo Swift
//
//  Created by Eduardo Mello de Macedo | Stone on 21/03/17.
//  Copyright © 2017 Eduardo Mello de Macedo | Stone. All rights reserved.
//

import UIKit
import StoneSDK

class TestValidationViewController: UIViewController {

    @IBOutlet weak var feedbackLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.navigationItem.title = "Testando Validações"
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    @IBAction func performActivation(_ sender: Any) {
        if (STNValidationProvider.validateActivation()){
            NSLog("Stone Code está ativado!")
            self.feedbackLabel.text = "Stone Code está ativado!"
        }else{
            NSLog("Stone Code não ativado.")
            self.feedbackLabel.text = "Stone Code não ativado."
        }
    }
    
    @IBAction func performPinpadConnection(_ sender: Any) {
        if (STNValidationProvider.validatePinpadConnection()){
            NSLog("O pinpad está pareado com o dispositivo iOS!")
            self.feedbackLabel.text = "O pinpad está pareado com o dispositivo iOS!"
        } else {
            NSLog("O pinpad não pareado com o dispositivo iOS!")
            self.feedbackLabel.text = "O pinpad não pareado com o dispositivo iOS!"
        }
    }
    @IBAction func performTableDownloaded(_ sender: Any) {
        if (STNValidationProvider.validateTablesDownloaded()) {
            NSLog("As tabelas já foram baixadas para o dispositivo iOS!");
            self.feedbackLabel.text = "As tabelas já foram baixadas para o dispositivo iOS!";
        } else {
            NSLog("As tabelas ainda não foram baixadas para o dispositivo iOS!");
            self.feedbackLabel.text = "As tabelas ainda não foram baixadas para o dispositivo iOS!";
        }
    }
    
    @IBAction func performConnectionNetwork(_ sender: Any) {
        if (STNValidationProvider.validateConnectionToNetWork()) {
            NSLog("A conexão com a internet está ativa!");
            self.feedbackLabel.text = "A conexão com a internet está ativa!";
        } else {
            NSLog("A conexão com a internet está inativa!");
            self.feedbackLabel.text = "A conexão com a internet está inativa!";
        }
    }
}
